src/main/java/org/broadinstitute/hellbender/tools/exome/ReadCountCollection.java

```diff
@@ -260,6 +262,37 @@ public ReadCountCollection subsetColumns(final Set<String> columnsToKeep) {
     }
 
     /**
+     * Rearrange the targets so that they are in a particular order.
+     * @return a new collection.
+     * @throws IllegalArgumentException if any of the following is true:
+     * <ul>
+     *     <li>{@code targetsInOrder} is {@code null},</li>
+     *     <li>is empty,</li>
+     *     <li>it contains {@code null},</li>
+     *     <li>contains any target not present in this collection.</li>
+     * </ul>
+     */
+    public ReadCountCollection arrangeTargets(final List<Target> targetsInOrder) {
+        Utils.nonNull(targetsInOrder);
+        if (targetsInOrder.isEmpty()) {
+            throw new IllegalArgumentException(String.format("the input targets list cannot be empty"));
+        }
+        final RealMatrix counts = new Array2DRowRealMatrix(targetsInOrder.size(), columnNames.size());
+        final Object2IntMap<Target> targetToIndex = new Object2IntOpenHashMap<>(targets.size());
```

Comment:
Is an more efficient map of Obj to primitive ints rather than Integers. Is amongst the dependencies so I don't see the harm in use this library for that.